src/math/mu_law.c

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2025 Intel Corporation.
+//
+// Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+
+#include <sof/audio/format.h>
+#include <sof/math/mu_law.h>
+#include <stdint.h>
+
+#define SOFM_MULAW_BIAS			33
+#define SOFM_MULAW_MAX			8191
+#define SOFM_MULAW_TOGGLE_BITS		0x7f
+#define SOFM_MULAW_SIGN_BIT		0x80
+#define SOFM_MULAW_MANTISSA_MASK	0x0f
+#define SOFM_MULAW_MANTISSA_BITS	4
+#define SOFM_MULAW_SHIFT_MASK		0x07
+#define SOFM_MULAW_DEC_ONES_MASK	0x21  /* 0b100001 for "1abcd1", see below */
+
+/*
+ * mu-law encode table (sign bit is b12)
+ *
+ * Input values 12:0		Output values 6:0
+ *
+ * 0 0 0 0 0 0 0 1 a b c d x      0 0 0 a b c d
+ * 0 0 0 0 0 0 1 a b c d x x      0 0 1 a b c d
+ * 0 0 0 0 0 1 a b c d x x x      0 1 0 a b c d
+ * 0 0 0 0 1 a b c d x x x x      0 1 1 a b c d
+ * 0 0 0 1 a b c d x x x x x      1 0 0 a b c d
+ * 0 0 1 a b c d x x x x x x      1 0 1 a b c d
+ * 0 1 a b c d x x x x x x x      1 1 0 a b c d
+ * 1 a b c d x x x x x x x x      1 1 1 a b c d
+ *
+ * A-law decode table (sign bit is b7)
+ *
+ * Input values 6:0	Output values 12:0
+ *
+ * 0 0 0 a b c d	0 0 0 0 0 0 0 1 a b c d 1
+ * 0 0 1 a b c d	0 0 0 0 0 0 1 a b c d 1 0
+ * 0 1 0 a b c d	0 0 0 0 0 1 a b c d 1 0 0
+ * 0 1 1 a b c d	0 0 0 0 1 a b c d 1 0 0 0
+ * 1 0 0 a b c d	0 0 0 1 a b c d 1 0 0 0 0
+ * 1 0 1 a b c d	0 0 1 a b c d 1 0 0 0 0 0
+ * 1 1 0 a b c d	0 1 a b c d 1 0 0 0 0 0 0
+ * 1 1 1 a b c d        1 a b c d 1 0 0 0 0 0 0 0
+ */
+
+/* Shift values lookup table for above table for 7
+ * highest sample value bits.
+ */
+static uint8_t mulaw_encode_shifts[128] = {
+	0, 1, 2, 2, 3, 3, 3, 3,
+	4, 4, 4, 4, 4, 4, 4, 4,
+	5, 5, 5, 5, 5, 5, 5, 5,
+	5, 5, 5, 5, 5, 5, 5, 5,
+	6, 6, 6, 6, 6, 6, 6, 6,
+	6, 6, 6, 6, 6, 6, 6, 6,
+	6, 6, 6, 6, 6, 6, 6, 6,
+	6, 6, 6, 6, 6, 6, 6, 6,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+	7, 7, 7, 7, 7, 7, 7, 7,
+};
+
+/**
+ * sofm_mu_law_encode() - Encode sample with A-law coding
```

Comment:
mu-law I suppose

Reply:
Oops, thanks! It's fixed now, and there was one more in addition to these.